Automattic/jetpack
# Beautiful Math: deprecate module and move code into shortcodes instead

#### Changes proposed in this Pull Request:

In many ways, the Beautiful Math module is just an embed like the other embed methods we make available in Shortcode Embeds. As we are re-thinking module organization in Jetpack (see #11931 and p1HpG7-6xr-p2), I thought it may be best if we just made that change.

This PR consequently deprecates the Beautiful Math module and introduces a LaTeX shortcode instead.

#### Notes

There are still things to do if we want to move ahead with this:

- [ ] I am not quite sure how we should handle sites that currently have the Beautiful Math module on, and the Shortcodes module off. This PR as it is would end up breaking all formulae on their site. Ideally we would display a notice on update, inviting them to activate the module?
- [ ] Move module support doc into a shortcode support doc: https://jetpack.com/support/beautiful-math-with-latex/
- [ ] Apply the same changes to WordPress.com
- [ ] update the Fusion list to bring the feature in sync.
- [ ] Handle conflicting plugins:

We used to check for other LaTeX plugins, and not auto-activate our module when another LaTeX plugin was already installed. 
I removed that code since we don't have a module anymore, and I've considered adding a similar short-circuit in the shortcode itself like so:

```php
// Get list of active plugins on the site.
$active_plugins = Jetpack::get_active_plugins();

// List of LaTeX plugins that may be used instead of our shortcode.
$latex_plugins = array(
	'latex/latex.php',
	'youngwhans-simple-latex/yw-latex.php',
	'easy-wp-latex-lite/easy-wp-latex-lite.php',
	'mathjax-latex/mathjax-latex.php',
	'enable-latex/enable-latex.php',
	'wp-quicklatex/wp-quicklatex.php',
);

if (
	! empty( $active_plugins )
	&& in_array( $latex_plugins, $active_plugins, true )
) {
	// Shortcircuit.
}
```
I did not make that change though as I am not sure it's really useful compared to the cost of checking for active plugins every time. What's your take on this?

#### Testing instructions:

* Try one of the embed methods at the top of the shortcode file and make sure Beautiful Math still works.
`$latex e^{\i \pi} + 1 = 0$`
`$latex [a, b]$`
`[latex s=4 bg=00f fg=ff0]\LaTeX[/latex]`

#### Proposed changelog entry for your changes:

* Beautiful Math: deprecate module and make it one of our shortcodes instead
